grammar.js

```diff
@@ -10,27 +11,30 @@ module.exports = grammar({
     // TODO: add the actual grammar rules
     source_file: $ =>
       seq(
-        $.preamble,
-        repeat($._definition)
+        optional($.control_section),
+        optional($.metadata_section),
+        optional($.shape_section)
       ),
 
+    shape_section: $ => seq(
+      $.namespace_statement,
+      repeat($._definition)
+    ),
+
     _definition: $ => choice(
       $.use_statement,
-      seq($.trait_statements, $.shape_statement),
```

Comment:
I removed this choice and updated `shape_statement` to include `optional($.trait_statements)`.